Src/ILGPU/Backends/SPIRV/ISPIRVBuilder.tt

```diff
@@ -0,0 +1,58 @@
+<#@ template hostspecific="true" language="C#" #>
+<#@ output extension=".cs"#>
+<#@ include file="BuilderCommon.ttinclude"
+#><#@ assembly name="System.Xml" #>
+<#@ assembly name="System.Core"#>
+<#@ import namespace="System.Linq"#>
+using System;
+using System.Collections.Generic;
+
+#nullable enable
+#pragma warning disable 1591
+
+namespace ILGPU.Backends.SPIRV {
+
+<#
+    PushIndent(standardIndent);
+#>
+/// <summary>
+/// Defines utility methods to generate SPIRV operations
+/// </summary>
+[CLSCompliant(false)]
+public interface ISPIRVBuilder
+{
+
```

Comment:
nit: remove empty line.
Question: do we really need to declare the whole interface in the T4 template? We can refine the code to define a common base interface as a partial interface in another C# file.

Reply:
So like move this section
```csharp
byte[] ToByteArray();

void AddMetadata(uint magic, uint version, uint genMagic, uint bound, uint schema);

// This is the best way I could come up with to
// handle trying to merge different builders
// Implementing classes will kinda just have to
// deal with it
void Merge(ISPIRVBuilder other);
```
to its own file? I guess we could do that but I'm not sure if it helps that much. If we do go forward with it we should do the same with all the other classes.